### ISSUE TITLE
Adding conversion_goals with a user_action goal results in Internal Server Error 500

### ISSUE THREAD

**Issue by TigerC10** [dynatrace-oss/terraform-provider-dynatrace#117]
Added the following block to my web application:
```hcl
  conversion_goals {
    goal {
      name = "Converted"
      type = "UserAction"

      user_action {
        action_type    = "Load"
        case_sensitive = false
        match_entity   = "ActionName"
        match_type     = "Contains"
        value          = "loading of page /conversion-page"
      }
    }
  }
```

And got this response:
```plain
Error: {
│   "code": 500,
│   "message": "Internal Server Error occurred. It has been logged and will be investigated.",
│   "constraintViolations": null
│ }
```

This seems like an issue with the API.  However, the one thing that I can think about that it might be an issue with the TF provider is the `id` field on the `goal` block.  I would assume the id should be generated by the API request.  But, I wonder if the marshaller is trying to send a null to the API.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hellot @TigerC10 ,

The underlying issue is the property "caseSensitive" for the UserActionDetails.
This property is being treated as "optional" by the provider - while the REST API insists that it's getting specified.

I'm working hard to push out a release that fixes a couple of bugs that have built up in the past by the end of this week. Your bugfix will be included.

**Comment by Dynatrace-Reinhard-Pilz** [comment]
This issue has been fixed with [Release v1.11.0](https://github.com/dynatrace-oss/terraform-provider-dynatrace/releases/tag/v1.11.0).
